python/ray/experimental/state.py

```diff
@@ -345,10 +348,22 @@ def task_profiles(self):
         executions of that task. The second element is a list of profiling
         information for tasks where the events have no task ID.
     """
+    if start == None:
+      start = 0
+    if end == None:
+      end = time.time()
+    if num == None:
+      num = sys.maxsize
     task_info = dict()
     event_names = self.redis_client.keys("event_log*")
+    start_window = start
+    end_window = end
+    smallest_heap = []
+    heapq.heapify(smallest_heap)
+    smallest_heap_size = 0
     for i in range(len(event_names)):
-      event_list = self.redis_client.lrange(event_names[i], 0, -1)
+      event_list = self.redis_client.zrangebyscore(event_names[i], min=start_window,
+                                                  max=end_window, start=0, num=num)
```

Comment:
We shouldn't need to provide the `start=0` argument, right? How does the `start` argument differ from the `min` argument?